ospf6d/ospf6_message.c

```diff
@@ -1709,6 +1709,14 @@ static int ospf6_read_helper(int sockfd, struct ospf6 *ospf6)
 		return OSPF6_READ_CONTINUE;
 	}
 
+	/*
+	 * Drop packet destined to another VRF if received in the default VRF.
+	 * This happens when raw_l3mdev_accept is set to 1.
+	 */
+	if (ospf6->vrf_id == VRF_DEFAULT
+	    && ospf6->vrf_id != oi->interface->vrf_id)
```

Comment:
why is `if (ospf6->vrf_id != oi->ionterface->vrf_id)` insufficient here?  I am not sure why we would need to test against VRF_DEFAULT

Reply:
Right, I also thought about that. But I decided to do the same thing you did in OSPFv2 to keep code consistent. How about changing both?

Reply:
go for it

Reply:
I'll change OSPFv2 in a separate PR.